### PR TITLE
listen for file changes instead of calling refresh manually in each c…

### PR DESCRIPTION
**Pull request by lhein**
…ommand #27

Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -139,6 +143,19 @@ function handleVisibleEditorChanges(event) {
...
+	fileSystemWatcher = vscode.workspace.createFileSystemWatcher('**/*.yaml');
+	fileSystemWatcher.onDidCreate( () => {
+		dataVirtProvider.refresh();
+	});
+	fileSystemWatcher.onDidDelete( () => {
+		dataVirtProvider.refresh();
+	});
+	fileSystemWatcher.onDidChange( () => {
+		dataVirtProvider.refresh();
+	});
```
Comment (apupier) [inline src/extension.ts:156]: I fear that a refresh for all files on the whole tree for all changes will cause troubles with big workspaces.

I think we should try with a relatively medium workspace at least (50 yaml files?) and see the impact. (or maybe you already tried?)
If nto noticeable, we migth go like that and create an enhancement request for later on real big workspaces. if it is causing delay, I think we should improve it within this PR

Comment (lhein) [reply src/extension.ts:156]: @rareddy What do you think? How many yaml files will the user have in the workspace?

Comment (lhein) [reply src/extension.ts:156]: I will create a test environment with at least 50 files. That said after discussion with Ramesh it turned out that the yaml files can be really huge. So we need to refactor the treemodel and logic anyways to ensure we are performing good enough. Raised https://issues.redhat.com/browse/FUSETOOLS2-344 for this purpose.

Comment (lhein) [reply src/extension.ts:156]: Tested with 50 yaml files and there was no difference in speed recognizable. That might become an issue later but for now I don't see this as a blocker.

**Comment by lhein** [comment]
Regarding the test of the refresh call. I cannot register a spy on the `dataVirtProvider`. If I try this it simply tells me on execution that it is `undefined` which means the activate function was never executed. 

`
 1) DataVirt Tree Provider Tests
       "before all" hook for "should call refresh when adding a new VDB":
     TypeError: Cannot read property 'refresh' of undefined
  	at Function.spy (node_modules/sinon/lib/sinon/spy.js:157:61)
  	at Sandbox.spy (node_modules/sinon/lib/sinon/sandbox.js:326:35)
  	at Context.<anonymous> (out/test/extension.test.js:9:3290)
  	at processImmediate (internal/timers.js:439:21)

base.js:266
  2) DataVirt Tree Provider Tests
       "after all" hook for "should call refresh when adding a new VDB":
     TypeError: Cannot read property 'restore' of undefined
  	at Context.<anonymous> (out/test/extension.test.js:9:3440)
  	at processImmediate (internal/timers.js:439:21)

`

**Comment by lhein** [comment]
That said all my modification tests for commands do not call the direct commands but rather some underlying functions which omits the refresh call to the provider. I think this should be tested in the UI tests instead.

**Comment by apupier** [comment]
> That said all my modification tests for commands do not call the direct commands but rather some underlying functions which omits the refresh call to the provider.

it is possible to add tests that are calling the commands themselves. Both tests makes sense.

> I think this should be tested in the UI tests instead.

that's possible too. Can you point to the issue that is covering this use case?

**Comment by apupier** [comment]
> the activate function was never executed.

in VS Code Camel K, there is an Utils method to ensure the extensions is activated before playing test: https://github.com/camel-tooling/vscode-camelk/blob/d6f334afcd0a3b8e1e963800e3028779090ee00f/src/test/suite/Utils.ts#L24-L34

**Pull request merged**